### PR TITLE
DM-46061: Add Metrics to calibrate task metadata

### PR DESCRIPTION
**Pull request by jrmullaney**

### CODE REVIEW

**Review by isullivan — CHANGES_REQUESTED** [review]
Please also add these metadata entries to `calibrateImage`, since we're trying to make the switch to the new task soon.

File: python/lsst/pipe/tasks/calibrate.py [hunk 1]
```diff
@@ -602,13 +604,19 @@ def run(self, exposure, background=None,
 
         detRes = self.detection.run(table=table, exposure=exposure,
                                     doSmooth=True)
+
+        self.recordFlaggedPixelFractions(exposure)
```
[the remark this answers is not in the document]
Comment (jrmullaney) [reply python/lsst/pipe/tasks/calibrate.py:607]: Given the issues that's happening with `calibrateImage` (see [this slack thread](https://lsstc.slack.com/archives/C2JPMCF5X/p1727991226934669)), I'd prefer to move this to another ticket until those issues have been resolved.

**Comment by jrmullaney** [comment]
You'll notice that I've also changed how values are written to metadata. It turns out that if one uses the `.add()` method, then it appends to a list of values under that name in the metadata. The catch is that the first value is written as a single-element list, which then gets read as an `ndarray` when the metadata is read-in by the analysis tool. This was the cause of the problem I mentioned on Slack. Using `self.metadata["some_new_metric"] = value` adds `value` as a scalar, rather than a list, which solves the issue.

**Review by isullivan — APPROVED** [review]
Good find about using `self.metadata["some_new_metric"] = value`. I have run into problems with metrics from `self.metadata.add` and this addresses those issues.

I am fine with postponing the additions to `calibrateImage`, but please create that ticket now and link it to DM-43077.

**Pull request merged**